Q 4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [3]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [7]:
#Load the data
df=pd.read_csv('C:/Users\ganes\Desktop\DS\Costomer+OrderForm.csv')
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [8]:
df.	Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [9]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [10]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [12]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [13]:
#Make the contingency/dimensional array
obs=np.array([[271, 267, 269, 280],[29,  33,  31,  20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [15]:
# Chi2 contingency independence test
chi2_contingency(obs) #(Chi2 stast value, p value, df, expected observation)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [ ]:
# Compare p_value with α = 0.05

Inerence: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre